src/coreclr/tests/src/Loader/binding/tracing/BinderTracingTest.DefaultProbing.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+using TestLibrary;
+
+namespace BinderTracingTests
+{
+    partial class BinderTracingTest
+    {
+        private static CultureInfo SatelliteCulture = CultureInfo.CreateSpecificCulture("fr-FR");
+
+        [BinderTest]
+        public static BindOperation AssemblyInAppPath()
+        {
+            AssemblyName assemblyName = new AssemblyName(DependentAssemblyName);
+            CustomALC alc = new CustomALC(nameof(AssemblyInAppPath));
+            Assembly asm = alc.LoadFromAssemblyName(assemblyName);
+
+            return new BindOperation()
+            {
+                AssemblyName = assemblyName,
+                AssemblyLoadContext = alc.ToString(),
+                Success = true,
+                ResultAssemblyName = asm.GetName(),
+                ResultAssemblyPath = asm.Location,
+                Cached = false,
+                ProbedPaths = new List<ProbedPath>()
+                {
+                    new ProbedPath()
+                    {
+                        FilePath = Helpers.GetProbingFilePath(ProbedPath.PathSource.AppNativeImagePaths, assemblyName.Name, isExe: false),
+                        Source = ProbedPath.PathSource.AppNativeImagePaths,
```

Comment:
This is unfortunate side-effect of  `corerun`... normal apps executed through apphost or muxer would not have these. Maybe add a comment describing why this works the way it does.